spec/couchbase/protostellar/collection_spec.rb

```diff
@@ -18,376 +18,14 @@
 require "couchbase/protostellar"
 
 RSpec.describe Couchbase::Protostellar::Collection do
-  let(:cluster) do
-    connect
-  end
-
-  let(:collection) do
-    default_collection(cluster)
-  end
-
-  let(:sample_content) do
-    {"value" => 42}
-  end
-
-  let(:updated_content) do
-    {"updated value" => 100}
-  end
+  subject(:collection) { @collection }
 
-  def upsert_sample_document(name: :foo, content: sample_content, doc_id: nil)
-    doc_id = unique_id(name) if doc_id.nil?
-    collection.upsert(doc_id, content)
-    doc_id
+  # rubocop:disable RSpec/BeforeAfterAll
+  before(:all) do
+    @cluster = connect("protostellar")
+    @collection = default_collection(@cluster)
```

Comment:
Not so sure about this here. Generally it seems that in RSpec instance variables are discouraged (Rubocop doesn't like it by default), and the use of `let`s encouraged. However, `let` statements are reevaluated in each example which seems like unnecessarily wasteful for things like cluster and collection objects, so I did this instead to only create the cluster connection once

Reply:
collections/scopes are rather light-weight object, at least in Classic SDK, they just wrap cluster with some extra options. I cannot think of a better solution here.

Reply:
Yes, they are very lightweight in protostellar as well. Clusters and buckets are also just simple objects in protostellar, they're not connecting to the server when they are initialised